### DISCUSSION TITLE
Mac/Mobile only?

### DISCUSSION THREAD

**Discussion by TonyDev314** [lexiforest/curl_cffi#459] [Q&A] [answered]
I'm using curl_cffi on Ubuntu in WSL.

Apart from the mobile browsers, all the others seem to give only Mac based headers - e.g. If I ask for latest Chrome I get
```
  "Sec-Ch-Ua-Platform": [
        "\"macOS\""
      ],
      "User-Agent": [
        "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/131.0.0.0 Safari/537.36"
      ],
```
    
Is this down to how it's compiled or is there a setting that I can also use Windows-based browsers?

**Comment by perkfly** [comment, marked as answer]
Just pass you own headers.

**Reply by TonyDev314** [reply]
Surely the whole point of this is to provide TLS that matches the headers.
If I start sending my own headers, they won't match.

**Reply by lexiforest** [reply]
Headers are not part of the TLS fingerprint, nor they are part of h2 fingerprint.

**Reply by TonyDev314** [reply]
Surely the TLS fingerprint is different when sent from a Mac than a Windows machine?

**Reply by lexiforest** [reply]
They are the same, because they use the exact same boringssl version, you can verify by wireshark or a TLS fingerprint website.
